crates/forge_services/src/tool_services/fs_read.rs

```diff
@@ -76,10 +76,13 @@ impl<F: FileInfoInfra + EnvironmentInfra + InfraFsReadService> FsReadService for
 
         let (start_line, end_line) = resolve_range(start_line, end_line, env.max_read_size);
 
-        let (content, file_info) = self
-            .0
-            .range_read_utf8(path, start_line, end_line)
-            .await
+        // Read the range for content and the entire file for hashing in parallel
+        let (range_result, full_content_result) = tokio::join!(
+            self.0.range_read_utf8(path, start_line, end_line),
+            self.0.read_utf8(path)
+        );
```

Comment:
If you are reading the whole file anyways, might as well slice and return a chunk of it.